egui/src/widgets/spinner.rs

```diff
@@ -0,0 +1,65 @@
+use epaint::{emath::lerp, vec2, Pos2, Shape, Stroke};
+
+use crate::{Response, Sense, Ui, Widget};
+
+/// A spinner widget used to indicate loading.
+#[must_use = "You should put this widget in an ui with `ui.add(widget);`"]
+pub struct Spinner {
+    enabled: bool,
+    /// Uses the style's `interact_size` if `None`.
+    size: Option<f32>,
+}
+
+impl Spinner {
+    /// A new spinner that is shown if `enabled` is true.
+    /// Uses the style's `interact_size` unless changed.
+    /// A disabled spinner still takes up it's usual space in order to prevent inconsistent
+    /// alignment.
+    pub fn new(enabled: bool) -> Self {
+        Self {
+            enabled,
+            size: None,
+        }
+    }
+
+    /// Sets the spinner's size. The size sets both the height and width, as the spinner is always
+    /// square.
+    pub fn size(mut self, size: f32) -> Self {
+        self.size = Some(size);
+        self
+    }
+}
+
+impl Widget for Spinner {
+    fn ui(self, ui: &mut Ui) -> Response {
+        let size = self
+            .size
+            .unwrap_or_else(|| ui.style().spacing.interact_size.y);
+        let (rect, response) = ui.allocate_exact_size(vec2(size, size), Sense::hover());
+
+        if self.enabled {
```

Comment:
```suggestion
        if ui.is_rect_visible(rect) {
```

Reply:
The idea of the enabled field was so that the spinner would still reserve the size it usually takes up, so that alignment is kept consistent with other widgets. Should I still change this?